opentelemetry-config/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+//! # Library for declarative configuration of OpenTelemetry.
+//!
+//! This library provides a way to configure OpenTelemetry SDK components
+//! using a declarative approach. It allows users to define configurations
+//! for metrics, traces, and exporters in a structured manner.
+
+use std::{
+    collections::HashMap,
+    error::{self, Error},
+    fmt::{self, Display},
+};
+
+use opentelemetry_sdk::{
+    logs::SdkLoggerProvider,
+    metrics::{MeterProviderBuilder, SdkMeterProvider},
+    trace::SdkTracerProvider,
+};
+use serde_yaml::Value;
+
+pub mod model;
+pub mod providers;
+
+/// Registry for different configuration providers.
+pub struct ConfigurationProvidersRegistry {
+    metrics: MetricsProvidersRegistry,
+}
+
+impl ConfigurationProvidersRegistry {
+    pub fn new() -> Self {
+        Self {
+            metrics: MetricsProvidersRegistry::new(),
+        }
+    }
+
+    pub fn metrics_mut(&mut self) -> &mut MetricsProvidersRegistry {
```

Comment:
This method is not needed for the reason mentioned in https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/487#discussion_r2532499511